### DISCUSSION TITLE
ImportError: 'IgniteInfo' from 'monai.utils' and Compatibility Issues with MONAI v1.4.0+ during Inference

### DISCUSSION THREAD

**Discussion by rifshu** [Project-MONAI/model-zoo#713] [Q&A] [answered]
I'm encountering multiple issues when running inference using the MONAI bundle on a custom pipeline that includes RetinaNet-based object detection for 3D medical imaging (LUNA16 dataset). The issues seem to be related to the recent updates in MONAI (v1.4.0 and above). Below are the details:

ImportError for IgniteInfo:
When attempting to run inference using the MONAI bundle (python -m monai.bundle run --config_file configs/inference.json), I receive the following error:

ImportError: cannot import name 'IgniteInfo' from 'monai.utils'

Expected Behavior:
The inference pipeline should run without errors using the pre-trained model and process the LUNA16 dataset to generate predictions.

Actual Behavior:
The pipeline fails to run due to IgniteInfo import errors, version incompatibilities with itkreader, and multiple deprecation warnings.

What I've Tried:
Updated all related packages (torch, pytorch-ignite, itk).
changes in inference file:
1) "whether_raw_luna16": true
2) "dataset_dir": "..../subset0",
Request for Help:
I am relatively new to MONAI and would greatly appreciate a step-by-step guide on how to properly use a pre-trained model from MONAI to run inference on a dataset that I have stored locally on my system. Any guidance on setting up the configuration files, loading the model, and executing the inference pipeline would be incredibly helpful.

Thank you in advance for your assistance!

**Comment by KumoLiu** [comment]
Hi @rifshu, may I know the version of this bundle you are using? Version 0.6.8 should already fix this issue. Could you please try with this version? Thanks.
https://github.com/Project-MONAI/model-zoo/blob/dev/models/lung_nodule_ct_detection/configs/metadata.json#L5

**Reply by rifshu** [reply]
Hi @KumoLiu , thanks for the quick response!

I am using the latest version of the MONAI bundle (0.6.8) as suggested. However, I am still encountering issues when running the scripts. Specifically, the scripts in my pipeline (like trainer.py and detection_saver.py) are still importing IgniteInfo from monai.utils, which leads to the following error:

ImportError: cannot import name 'IgniteInfo' from 'monai.utils'

 I think this error occurs because IgniteInfo has been removed in the latest versions of MONAI (v1.4.0 and above). While the metadata file indicates that version 0.6.8 of the bundle should address this, it seems that some parts of the scripts are not updated to reflect the changes.
(https://github.com/Project-MONAI/model-zoo/blob/dev/models/lung_nodule_ct_detection/scripts/detection_saver.py)

Additional Details:
I have ensured that I am using:
MONAI version: 1.4.0
MONAI bundle version: 0.6.8

I also tried updating all dependencies, clearing the cache, and using a fresh virtual environment, but the issue persists.

Would you be able to provide guidance on how to update these scripts or if there are specific files that need manual modifications to work with the latest MONAI version?

**Reply by KumoLiu** [reply, marked as answer]
Hi @rifshu,
It seems you're not using the correct MONAI version. In MONAI v1.4, you can use either `from monai.config import IgniteInfo` or `from monai.utils import IgniteInfo`, and both should work.

Here's the version check:
```
>>> monai.__version__
'1.4.0+13.g530cc1f3'
>>> from monai.utils import IgniteInfo
>>> from monai.config import IgniteInfo
```
You can update the file to use `from monai.config import IgniteInfo`. Alternatively, version 0.6.7 may also work in your environment. Another option is to check for multiple MONAI versions—uninstalling all versions and reinstalling 1.4 could resolve the issue.
https://github.com/Project-MONAI/model-zoo/blob/dev/models/lung_nodule_ct_detection/scripts/detection_saver.py#L18

**Reply by rifshu** [reply]
Thanks, Issue has been resolved.

**Comment by rifshu** [comment]
hi,

I have downloaded the resulted JSON file of dataset:
https://github.com/Project-MONAI/MONAI-extra-test-data/releases/download/0.8.1/LUNA16_datasplit-20220615T233840Z-001.zip

and in Inference.json file changes I have made are:

1)  "data_list_file_path": ***@***.***_root +
'/LUNA16_datasplit/dataset_fold0.json'",

2)  "dataset_dir": ".../_/LUNA16_datasplit/mhd_original/dataset_fold0.json",


error:
RuntimeError: LoadImage cannot find a suitable reader for file:
........\_\LUNA16_datasplit\mhd_original\dataset_fold0.json\1.3.6.1.4.1.14519.5.2.1.6279.6001.281489753704424911132261151767\1.3.6.1.4.1.14519.5.2.1.6279.6001.281489753704424911132261151767.nii.gz.

request:
* I have resolved some of the previous issues, but now I'm encountering a
problem related to the dataset. Could you please guide me on where I can
download the correct version of the LUNA16 dataset? Additionally, if I
already have access to the original LUNA16 dataset files (in .mhd and .raw
formats), what changes do I need to make in my inference.json file to
successfully run inference on this original dataset? Specifically, I am
looking for guidance on updating the paths, data transforms, or reader
configurations to ensure that MONAI can properly load and process the
original LUNA16 dataset for nodule detection inference. Any help would be
greatly appreciated!  *

**Reply by KumoLiu** [reply]
Hi @rifshu, you can find more details regarding the data info here: https://github.com/Project-MONAI/model-zoo/tree/dev/models/lung_nodule_ct_detection#data
